analytic_engine/src/instance/drop.rs

```diff
@@ -58,19 +54,21 @@ impl Dropper {
             return Ok(false);
         }
 
-        // Fixme(xikai): Trigger a force flush so that the data of the table in the wal
-        //  is marked for deletable. However, the overhead of the flushing can
-        //  be avoided.
-
-        let opts = TableFlushOptions::default();
-        let flush_scheduler = serial_exec.flush_scheduler();
-        self.flusher
-            .do_flush(flush_scheduler, &table_data, opts)
+        // Mark table's WAL for deletable, memtable will also get freed automatically
+        // when table_data is dropped.
+        let table_location = table_data.table_location();
+        let wal_location =
+            crate::instance::create_wal_location(table_location.id, table_location.shard_info);
+        // Use max to represent delete all WAL.
+        // TODO: add a method in wal_manager to delete all WAL with same prefix.
+        let sequence = MAX_SEQUENCE_NUMBER;
+        self.space_store
+            .wal_manager
+            .mark_delete_entries_up_to(wal_location, sequence)
```

Comment:
Maybe we should mark wal deleted after the table is dropped from the manifest.

Reply:
Added in comments.